### PR TITLE
Implement shared BrowserCompat shortcode

### LINKED ISSUE #33: Move over BrowserCompat widget
Folks want to use it on d.c.c. as well, so it should be moved into this repo.

### PR DESCRIPTION
**Pull request by matthiasrohmer**
Fixes #33. This ports over the BrowserCompat shortcode to be used by both web.dev and d.c.c. The following changes have been made:

- Tests are ported from mocha to ava
- Required dependencies are added to package (11ty for cache plugin, browser compat data)
- i18n functionality is ported over from web.dev and rewritten using fg (which already is a dependency from 11ty) instead of custom file walking. This also allows loading i18n from multiple sources, to be able to load translations from webdev-infra and d.c.c/web.dev in parallel
- Shortcode is rewritten to be templatable: you can configure a njk template instead of mixing the HTML into your shortcode JS. This allows reusing the logic on both web.dev and d.c.c but having separate markup/styling.

### CODE REVIEW

**Comment by matthiasrohmer** [comment]
It's been a while since I push myself to pick something up again, sorry it took so long. 😄 But I now feel good with where we ended up. Let me know what you think!

The prefix I choose for classes is `.wdi-`, happy to change. The fallback colors are taken from web.dev and also work for d.c.c. but would now be technically configurable by doing

```scss
.wdi-browser-compat {
  --wdi-text-color: pink;
}
```

in either the web.dev or the d.c.c repo.

Not ideal: browser icons were implemented as CSS background images before, they are now added to the markup as otherwise we would need a way to copy assets from webdev-infra to dist which seemed wrong. On the other hand that also saves some requests 🙂

File: filters/i18n.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  configure(config) {
+    if (!config.defaultLocale) {
+      throw new Error('[i18nFilter] No default locale given.');
+    }
+
+    this.defaultLocale = config.defaultLocale;
+
+    if (Array.isArray(config.dictPaths)) {
+      for (const path of config.dictPaths) {
+        dictionary.load(path);
```
Comment (devnook) [inline filters/i18n.js:37]: dictionary is an instance of Dictionary, so it already has some loaded data. It looks like you load it twice?

**Comment by devnook** [comment]
Regarding icons, I think it is good that they live as markup in this repo, esp that they are not dcc or web.dev dependent.

File: utils/dictionary.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+const fs = require('fs');
+const path = require('path');
+
+const fg = require('fast-glob');
+const yaml = require('js-yaml');
+const get = require('lodash.get');
+const set = require('lodash.set');
+
+class Dictionary {
```
[the remark this answers is not in the document]
Comment (matthiasrohmer) [reply utils/dictionary.js]: Good suggestion, done!

File: filters/i18n.js (hunk 1 above)
Comment (matthiasrohmer) [reply filters/i18n.js]: The idea is for this to replace the diverging implementations of the i18n util on DCC and web.dev, which could then load their own translation via this API.

**Review by devnook — APPROVED** [review]
Please consider adding a comment in the css file mentioning performance reasons.

**Pull request merged**